src/core/analysis-context.ts

```diff
@@ -483,19 +485,26 @@ export class AnalysisContext {
     }
   }
 
-  /**
-   * Returns true if the url given is resovable by the Analyzer's `UrlResolver`.
-   */
-  canResolveUrl(url: PackageRelativeUrl): boolean {
-    return this.resolver.canResolve(url);
-  }
-
   /**
    * Resolves a URL with this Analyzer's `UrlResolver` or returns the given
    * URL if it can not be resolved.
    */
-  resolveUrl(url: PackageRelativeUrl): ResolvedUrl {
-    return this.resolver.canResolve(url) ? this.resolver.resolve(url) :
-                                           (url as any as ResolvedUrl);
+  resolveUrl(url: PackageRelativeUrl): ResolvedUrl|undefined {
+    const resolved = this.resolver.resolve(url);
+    if (resolved === undefined) {
+      return url as any as ResolvedUrl;
+    }
+    return resolved;
+  }
+
+  resolveUrls(urls: PackageRelativeUrl[]): ResolvedUrl[] {
+    const results = [];
+    for (const url of urls) {
+      const resolved = this.resolveUrl(url);
+      if (resolved !== undefined) {
```

Comment:
Why the check for `undefined`? `resolveUrl` returns the input if the resolver can't resolve, so this is just filtering `undefined` as an input URL. Did you mean to filter out URLs that couldn't be resolved instead?